plugins/movetoarchive.koplugin/main.lua

```diff
@@ -0,0 +1,161 @@
+local ConfirmBox = require("ui/widget/confirmbox")
+local DataStorage = require("datastorage")
+local DocSettings = require("docsettings")
+local ReadCollection = require("readcollection")
+local ReadHistory = require("readhistory")
+local ReaderUI = require("apps/reader/readerui")
+local FileManager = require("apps/filemanager/filemanager")
+local InfoMessage = require("ui/widget/infomessage")
+local LuaSettings = require("frontend/luasettings")
+local UIManager = require("ui/uimanager")
+local WidgetContainer = require("ui/widget/container/widgetcontainer")
+local logger = require("logger")
+local util = require("frontend/util")
+local BaseUtil = require("ffi/util")
+local _ = require("gettext")
+
+local MoveToArchive = WidgetContainer:new{
+    name = "move2archive",
+    move_to_archive_settings = LuaSettings:open(("%s/%s"):format(DataStorage:getSettingsDir(), "move_to_archive_settings.lua")),
```

Comment:
You can move this one in` MoveToArchive:init()`, that's where I think this should be done.